spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/KinesisMessageChannelBinder.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
+import org.springframework.cloud.stream.binder.ExtendedProducerProperties;
+import org.springframework.cloud.stream.binder.ExtendedPropertiesBinder;
+import org.springframework.cloud.stream.binder.kinesis.properties.KinesisBinderConfigurationProperties;
+import org.springframework.cloud.stream.binder.kinesis.properties.KinesisConsumerProperties;
+import org.springframework.cloud.stream.binder.kinesis.properties.KinesisExtendedBindingProperties;
+import org.springframework.cloud.stream.binder.kinesis.properties.KinesisProducerProperties;
+import org.springframework.cloud.stream.binder.kinesis.provisioning.KinesisTopicProvisioner;
+import org.springframework.cloud.stream.provisioning.ConsumerDestination;
+import org.springframework.cloud.stream.provisioning.ProducerDestination;
+
+import com.amazonaws.services.kinesis.AmazonKinesisAsync;
+import com.amazonaws.services.kinesis.AmazonKinesisAsyncClientBuilder;
+
+import org.springframework.integration.core.MessageProducer;
+import org.springframework.integration.metadata.MetadataStore;
+import org.springframework.integration.metadata.SimpleMetadataStore;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.integration.aws.inbound.kinesis.CheckpointMode;
+import org.springframework.integration.aws.inbound.kinesis.KinesisMessageDrivenChannelAdapter;
+import org.springframework.integration.aws.inbound.kinesis.KinesisShardOffset;
+import org.springframework.integration.aws.inbound.kinesis.ListenerMode;
+import org.springframework.integration.aws.outbound.KinesisMessageHandler;
+import org.springframework.integration.channel.QueueChannel;
+
+/**
+ * 
+ * @author Peter Oates
+ */
+public class KinesisMessageChannelBinder extends
+		AbstractMessageChannelBinder<ExtendedConsumerProperties<KinesisConsumerProperties>, ExtendedProducerProperties<KinesisProducerProperties>, KinesisTopicProvisioner>
+		implements ExtendedPropertiesBinder<MessageChannel, KinesisConsumerProperties, KinesisProducerProperties> {
+
+	// not using currently but expect to include global properties here if
+	// required
+	private final KinesisBinderConfigurationProperties configurationProperties;
+
+	// not currently utilising any extended properties - not sure if I will need
+	// these
+	private KinesisExtendedBindingProperties extendedBindingProperties = new KinesisExtendedBindingProperties();
+
+	private AmazonKinesisAsync amazonKinesis;
+
+	public KinesisMessageChannelBinder(KinesisBinderConfigurationProperties configurationProperties,
+			KinesisTopicProvisioner provisioningProvider) {
+		super(false, null, provisioningProvider);
+		this.configurationProperties = configurationProperties;
+
+		this.amazonKinesis = AmazonKinesisAsyncClientBuilder.defaultClient();
```

Comment:
I think we have to include `<artifactId>spring-cloud-aws-autoconfigure</artifactId>` dependency and make this instance based on the auto-configured `credentials` and `region`:
```
AmazonKinesisAsyncClientBuilder
    .standard()
    .withCredentials(...)
    .withRegion(...)
```